### DISCUSSION TITLE
Is it possible to wait for a JS promise inside of applescript?

### DISCUSSION THREAD

**Discussion by J-Michalek** [johnlindquist/kit#1037] [Q&A]
My idea is basically to go to a website, then execute some async JS on the website using applescript and once the JS promise gets resolved return the resolved value back from the applescript so that I can work with it afterwards. So far I only figured out that the applescript returns results of synchronous JS.

example:
``` javascript
browse("someWebsite")

const js = `
 await (() => {
  const promise = new Promise((res) => {
  setTimeout(() => res('hello'), 500) })

  return promise
})()
`

const value = await applescript(`
tell application "Google Chrome" to tell window 1
	get execute active tab javascript "

${js}

"
end tell
`)

// Continue working with the value returned from applescript.
```

**Comment by johnlindquist** [comment]
Hmm, I'll have to dig into this sometime. I've never tried it.

We are planning on building Script Kit extension for Chrome which should allow us much, much more control, so that might ultimately be the answer.

**Reply by J-Michalek** [reply]
Sounds good, for my use case I can work with Selenium to achieve what I'm looking for. Also I'm really enjoying the app.
